rsearch/client.go

```diff
@@ -0,0 +1,47 @@
+package rsearch
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"log"
+	"net/http"
+)
+
+// SearchResource connects to instance of a server
+// and resolves SearchRequest
+func SearchResource(config Config, req SearchRequest) SearchResponse {
+	// TODO need to make url configurable
+	url := config.Server.Host + ":" + config.Server.Port
+	data := []byte(`{ "tag" : "` + req.Tag + `"}`)
+	if config.Server.Debug {
+		log.Println("Making request with", string(data))
+	}
+
+	// Make request
+	request, err := http.NewRequest("POST", url, bytes.NewBuffer(data))
+	request.Header.Set("Content-Type", "application/json")
+	client := &http.Client{}
+	response, err := client.Do(request)
```

Comment:
might as well use `http.DefaultClient` if you're not doing any special setup of the client or transport.